File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/legacy/mod11/contract/erp/BaseErpContractTest.java
package com.mycompany.myshop.systemtest.legacy.mod11.contract.erp;

import com.mycompany.myshop.systemtest.legacy.mod11.contract.base.BaseExternalSystemContractTest;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

public abstract class BaseErpContractTest extends BaseExternalSystemContractTest {
    @Test
    void shouldBeAbleToGetProduct() {
        assertDoesNotThrow(() -> scenario
                .given().product().withSku("SKU-123").withUnitPrice(12.0)
                .then().product("SKU-123").hasSku("SKU-123").hasPrice(12.0));
    }

}

File: system-test/dotnet/SystemTests/Legacy/Mod11/ExternalSystemContractTests/Erp/BaseErpContractTest.cs
using SystemTests.Legacy.Mod11.ExternalSystemContractTests.Base;

namespace SystemTests.Legacy.Mod11.ExternalSystemContractTests.Erp;

public abstract class BaseErpContractTest : BaseExternalSystemContractTest
{
    [Fact]
    public async Task ShouldBeAbleToGetProduct()
    {
        (await Scenario()
            .Given().Product().WithSku("BOOK-123").WithUnitPrice(12.0m)
            .Then().Product("BOOK-123"))
            .HasSku("BOOK-123")
            .HasPrice(12.0m);
    }
}










